### DISCUSSION TITLE
How to best annotate class methods as computeds?

### DISCUSSION THREAD

**Discussion by Obiwarn** [mobxjs/mobx#3052] [Q&A] [answered]
We do not want to use getters ([AirBnb Style Guide](https://github.com/airbnb/javascript#accessors--no-getters-setters)).

I.e. instead of `get val` we use `getVal()`.

I tried to understand from the docs, but I am not sure. Am I supposed to annotate computeds like this? Is there a more elegant way?

```
class Foo {
  observedVar = "Foo";
  getBar = computed(()=> {return this.observedVar })
}
```

Thank you.

**Comment by kubk** [comment]
> Do not use JavaScript getters/setters as they cause unexpected side effects and are harder to test, maintain, and reason about.

This paragraph is completely baseless. A getter it's just syntax sugar for a class method. There is no difference between testing a regular method and getter. An elegant way is to use getters :)

**Reply by Obiwarn** [reply]
This is indeed a matter of taste. 

However, if the existing codebase avoids getters, is the only way to annotate the above notation?

**Comment by mweststrate** [comment, marked as answer]
Yes
